### ISSUE TITLE
Exception be raised when I commit.

### ISSUE THREAD

**Issue by lagelalegal** [sebdah/git-pylint-commit-hook#50]
```
fatal: Unable to create 'path/.git/index.lock': File exists.

If no other git process is currently running, this probably means a
git process crashed in this repository earlier. Make sure no other git
process is running and remove the file manually to continue.
fatal: Unable to create '/Users/*/.git/index.lock': File exists.

If no other git process is currently running, this probably means a
git process crashed in this repository earlier. Make sure no other git
process is running and remove the file manually to continue.
Index was not unstashed.
Traceback (most recent call last):
  File "/usr/local/bin/git-pylint-commit-hook", line 4, in <module>
    __import__('pkg_resources').run_script('git-pylint-commit-hook==2.1.1', 'git-pylint-commit-hook')
  File "/usr/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 726, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1651, in run_script
    exec(code, namespace, namespace)
  File "/usr/local/lib/python2.7/site-packages/git_pylint_commit_hook-2.1.1-py2.7.egg/EGG-INFO/scripts/git-pylint-commit-hook", line 69, in <module>
    main()
  File "/usr/local/lib/python2.7/site-packages/git_pylint_commit_hook-2.1.1-py2.7.egg/EGG-INFO/scripts/git-pylint-commit-hook", line 62, in main
    args.limit, args.pylint, args.pylintrc, args.pylint_params, args.suppress_report)
  File "/usr/local/lib/python2.7/site-packages/git_pylint_commit_hook-2.1.1-py2.7.egg/git_pylint_commit_hook/commit_hook.py", line 263, in check_repo
    i += 1
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 24, in __exit__
    self.gen.next()
  File "/usr/local/lib/python2.7/site-packages/git_pylint_commit_hook-2.1.1-py2.7.egg/git_pylint_commit_hook/commit_hook.py", line 147, in _stash_unstaged
    subprocess.check_call('git stash pop --index -q'.split())
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['git', 'stash', 'pop', '--index', '-q']' returned non-zero exit status 128
```

**Comment by lagelalegal** [comment]
The project path, I'm mock it.

**Comment by evanunderscore** [comment]
Do you remember what happened to cause that file to be created in the first place? Was that related to the commit hook or did it happen separately? The operation that failed appears to be `git stash pop`, which is the last thing it does. I don't understand how the `git reset` immediately before it worked if this failed.

**Comment by sebdah** [comment]
A fix has been merged and released in 2.2.0
